src/services/cart-service.ts

```diff
@@ -0,0 +1,40 @@
+import { CartItem } from "../entities/cart-item";
+
+export default class CartService {
+  static addToCart(cartItem: CartItem, cart: CartItem[]): CartItem[] {
+    const existingItem = cart.find(
+      (item) => item.product.id === cartItem.product.id,
+    );
+
+    if (existingItem)
+      return this.updateCartItemQuantity(existingItem, cart, cartItem.quantity);
+    else return [...cart, cartItem];
+  }
+  static removeFromCart(cartItem: CartItem, cart: CartItem[]): CartItem[] {
+    return cart.filter((item) => item.product.id != cartItem.product.id);
+  }
+
+  static updateCartItemQuantity(
+    cartItem: CartItem,
+    cart: CartItem[],
+    newQuantity: number,
+  ): CartItem[] {
+    const existingItem = cart.find(
```

Comment:
There is no need to extract the existingItem, you can write like this:

```
if (newQuantity != 0)
      return cart.map((item) =>
        item.product.id === cartItem.product.id
          ? { ...existingItem, quantity: newQuantity }
          : item,
      );
    else if ... 
```

Reply:
Indeed, thank you for pointing it out!
You probably mean to use the spread operator on `cartItem`.
Here I think I duplicated the `addToCart` function and forgot to adjust the logic for the update function, only making essential changes.